# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsIDragService.idl
# Commit: df85fe3d7cb6
# Full Hash: df85fe3d7cb6a219c37e0a80d00f51e07783aec1
# Author: David P <daparks@mozilla.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 22 - Make nsIDragService::StartDragSession return the nsIDragSession r=rkraesig,win-reviewers,gstoll,geckoview-reviewers,m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D212298
# ==============================================================================

diff -r 384b025f56d3 -r df85fe3d7cb6 widget/nsIDragService.idl
--- a/widget/nsIDragService.idl	Thu Jul 04 07:48:11 2024 +0000
+++ b/widget/nsIDragService.idl	Thu Jul 04 07:48:12 2024 +0000
@@ -143,9 +143,10 @@
     * @param aWidgetProvider Either the nsIWidget that will be the target of
     *                        the drag, or an nsIDOMWindowProxy for a
     *                        window that is on that widget.
+    * @returns the new (or pre-existing) session
     */
-  [noscript]
-  void startDragSession(in nsISupports aWidgetProvider);
+  [notxpcom, nostdcall]
+  nsIDragSession startDragSession(in nsISupports aWidgetProvider);
 
   /**
    * Similar to startDragSession(), automated tests may want to start